scheduler.go

```diff
@@ -342,6 +342,8 @@ func (s *Scheduler) removeByCondition(shouldRemove func(*Job) bool) {
 	for _, job := range s.Jobs() {
 		if !shouldRemove(job) {
 			retainedJobs = append(retainedJobs, job)
+		} else {
+			job.timer.Stop()
```

Comment:
the timer may be nil here if the job has never been started. I would suggest adding a function to the job file like:

```
func (j *Job) stopTimer() {
	if j.timer != nil {
		j.timer.Stop()
	}
}
```

and the call that function instead of `job.timer.Stop()` as `job.stopTimer()`